docs/adr-dependencies.md

```diff
@@ -0,0 +1,86 @@
+# usage of Dependencies, peerDependencies, devDependencies in package.json
+
+## Context
+
+Dependencies are specified using the package.json of libraries.
+We have more and more libraries written and used by projects.
+
+We use different distribution format for our libraries:
+
+- UMD for some (react-components, react-cmf, etc ...)
+- mix of transpiled js and sass files which need shared config to make it work for others
+
+## Problems
+
+We see commits with mix of peerDependencies and dependencies without real guidelines.
+Each change in peerDependencies lead to a major release.
+
+We do not know when to put a dependencies in `"peerDependencies"` or just in `"dependencies"` attribute of the package.json
+
+With yarn peerDependencies are hard to manage. It just does a warning the first time , when it needs to really install it in your node_modules. So it is easy to forget it. We have also seen some false positive warnings.
+
+With npm it depends the version, the behavior change:
+
+- npm 1,2,7,8 install peerDependencies
+- npm 3,4,5,6 do not install peerDependencies
+
+The documentation of npm only cover the behavior of npm note the definition and usage of theses.
+
+We know our project can't use npm > 7.0 because we have some issues in our peerDependencies definitions ! That need to be fixed.
+
+## Solutions
+
+Define a clear guideline and usage of when and how to write the requirements in package.json of libraries.
+
+### Guideline
+
+**`"devDependencies"`**
+
+Add **a-dependency** under devDependencies if **a-dependency** is used only to build, debug, or because A is also a peerDependencies.
```

Comment:
```suggestion
Add **a-dependency** under `devDependencies` if **a-dependency** is used only to build, debug, or because A is also a peerDependencies.

Basically use `devDependencies` if this dependency has no impact on runtime.
```

> because A is also a peerDependencies

I don't understand this part!

Reply:
If you put react in peerDependency, you have also to put it in devDependencies or it will no be installed by many package manager (yarn, npm 3 to 6)